### PR TITLE
RBFPolicy Cleanup

### PR DESCRIPTION
**Pull request by stratisMarkou**
Renamed RBFPolicy to EQPolicy and dependencies and other changes:

- Moved the hacky assignments out of `EQPolicy.reset()`. Made a `PendulumEQPolicy` class which inherits `EQPolicy` and overrides the `reset` with the pendulum-specific values. I've kept this in `eq_policy.py` but we can move it.
- Now that `EQPolicy` is more general, I've updated the unit test to loop over and check different state dims, action dims and numbers of EQ features.
- Moved the pendulum example scripts, gifs directory, plots directory and saved agents dir under `examples/pendulum`. Maybe the `PendulumEQPolicy` should go in there too.

For future (added to readme):
- Think about better ways to do MC tests. Ideally, all `call`s should be able to handle batching so we don't have to have for-loops anywhere.
- Have some concept of a *feasible space* and/or *initialisation space* to do parameter resets more intelligently.

### CODE REVIEW

File: examples/pendulum/pendulum.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import numpy as np
 
 from pilco.environments import Environment
-from pilco.policies import RBFPolicy, TransformedPolicy
+from pilco.policies import PendulumEQPolicy, TransformedPolicy
```
Comment (gergely-flamich) [inline examples/pendulum/pendulum.py]: Yeah, as you said, it's probably cleaner to move PendulumEQPolicy to `examples/`. I think we should keep the contents of the package general.

File: pilco/policies/eq_policy.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        super().__init__(state_dim=state_dim,
+                         action_dim=action_dim,
+                         name=name,
+                         dtype=dtype,
+                         **kwargs)
+
+        # Number of radial basis functions
+        self.num_eq_features = num_eq_features
+
+        # Set RBF policy locations
```
Comment (gergely-flamich) [inline pilco/policies/eq_policy.py]: Change these comments too

File: pilco/policies/eq_policy.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+        eq_locs_init = tf.zeros((num_eq_features, state_dim), dtype=dtype)
+        self.eq_locs = UnconstrainedVariable(eq_locs_init,
+                                             name='eq_locs',
+                                             dtype=dtype)
```
Comment (gergely-flamich) [inline pilco/policies/eq_policy.py:36]: These variables will need to be re-written to use `ntfo.BoundedVariable`.

File: pilco/policies/eq_policy.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+        # RBF output is the weighted sum of eq components
+        eq = tf.einsum('i, i ->',
+                        self.eq_weights(),
+                        exp_quad)
+
+        return eq
+
+
+
+class PendulumEQPolicy(EQPolicy):
```
Comment (gergely-flamich) [inline pilco/policies/eq_policy.py]: Move this to its own file.

**Review by gergely-flamich — CHANGES_REQUESTED** [review]
I think you should make the small changes I indicated in the comments.

File: pilco/policies/eq_policy.py (hunk 3 above)
Comment (stratisMarkou) [reply pilco/policies/eq_policy.py:36]: Let's leave this for later.

**Review by gergely-flamich — APPROVED** [review]
Nice, merge when you are ready.

**Pull request merged**